x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -911,14 +923,45 @@ export class ObservabilityAIAssistantClient {
     return this.dependencies.knowledgeBaseService.deleteEntry({ id });
   };
 
-  private addResponseLanguage = (messages: Message[], responseLanguage: string): Message[] => {
+  private resolveInstructions = async (requestInstructions: UserInstruction[]) => {
+    const knowledgeBaseInstructions =
+      await this.dependencies.knowledgeBaseService.getInstructions();
+
+    if (requestInstructions.length + knowledgeBaseInstructions.length === 0) {
+      return '';
+    }
+
+    const knowledgeBaseInstructionsById = knowledgeBaseInstructions.reduce(
+      (acc, next) => ({
+        ...acc,
```

Comment:
can we avoid creating a new object here? maybe this is better as a simple `forEach`?

Reply:
Fixed in https://github.com/elastic/kibana/pull/180263/commits/341d83832b9ae69f8094bd21b7f7912dab3156ed